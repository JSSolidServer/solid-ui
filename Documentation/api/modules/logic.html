<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>logic | solid-ui</title>
	<meta name="description" content="Documentation for solid-ui">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="../assets/css/main.css">
	<script async src="../assets/js/search.js" id="search-script"></script>
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="../assets/js/search.json" data-base="..">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="../index.html" class="title">solid-ui</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="../modules.html">solid-ui</a>
				</li>
				<li>
					<a href="logic.html">logic</a>
				</li>
			</ul>
			<h1>Module logic</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<section class="tsd-panel-group tsd-index-group">
				<h2>Index</h2>
				<section class="tsd-panel tsd-index-panel">
					<div class="tsd-index-content">
						<section class="tsd-index-section ">
							<h3>Variables</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-variable tsd-parent-kind-module"><a href="logic.html#authn" class="tsd-kind-icon">authn</a></li>
								<li class="tsd-kind-variable tsd-parent-kind-module"><a href="logic.html#chat" class="tsd-kind-icon">chat</a></li>
								<li class="tsd-kind-variable tsd-parent-kind-module"><a href="logic.html#kb" class="tsd-kind-icon">kb</a></li>
								<li class="tsd-kind-variable tsd-parent-kind-module"><a href="logic.html#profile" class="tsd-kind-icon">profile</a></li>
								<li class="tsd-kind-variable tsd-parent-kind-module"><a href="logic.html#solidlogicsingleton" class="tsd-kind-icon">solid<wbr>Logic<wbr>Singleton</a></li>
								<li class="tsd-kind-variable tsd-parent-kind-module"><a href="logic.html#store" class="tsd-kind-icon">store</a></li>
							</ul>
						</section>
					</div>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Variables</h2>
				<section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module">
					<a name="authn" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> authn</h3>
					<div class="tsd-signature tsd-kind-icon">authn<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">AuthnLogic</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/solid/solid-ui/blob/89676eb/src/logic.ts#L14">src/logic.ts:14</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module">
					<a name="chat" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> chat</h3>
					<div class="tsd-signature tsd-kind-icon">chat<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">ChatLogic</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/solid/solid-ui/blob/89676eb/src/logic.ts#L15">src/logic.ts:15</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module">
					<a name="kb" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> kb</h3>
					<div class="tsd-signature tsd-kind-icon">kb<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">LiveStore</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/solid/solid-ui/blob/89676eb/src/logic.ts#L12">src/logic.ts:12</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module">
					<a name="profile" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> profile</h3>
					<div class="tsd-signature tsd-kind-icon">profile<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">ProfileLogic</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/solid/solid-ui/blob/89676eb/src/logic.ts#L19">src/logic.ts:19</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module">
					<a name="solidlogicsingleton" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> solid<wbr>Logic<wbr>Singleton</h3>
					<div class="tsd-signature tsd-kind-icon">solid<wbr>Logic<wbr>Singleton<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">SolidLogic</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/solid/solid-ui/blob/89676eb/src/logic.ts#L8">src/logic.ts:8</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module">
					<a name="store" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> store</h3>
					<div class="tsd-signature tsd-kind-icon">store<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">LiveStore</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/solid/solid-ui/blob/89676eb/src/logic.ts#L11">src/logic.ts:11</a></li>
						</ul>
					</aside>
				</section>
			</section>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class=" ">
						<a href="../modules.html">Exports</a>
					</li>
					<li class="current tsd-kind-module">
						<a href="logic.html">logic</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
					<li class=" tsd-kind-variable tsd-parent-kind-module">
						<a href="logic.html#authn" class="tsd-kind-icon">authn</a>
					</li>
					<li class=" tsd-kind-variable tsd-parent-kind-module">
						<a href="logic.html#chat" class="tsd-kind-icon">chat</a>
					</li>
					<li class=" tsd-kind-variable tsd-parent-kind-module">
						<a href="logic.html#kb" class="tsd-kind-icon">kb</a>
					</li>
					<li class=" tsd-kind-variable tsd-parent-kind-module">
						<a href="logic.html#profile" class="tsd-kind-icon">profile</a>
					</li>
					<li class=" tsd-kind-variable tsd-parent-kind-module">
						<a href="logic.html#solidlogicsingleton" class="tsd-kind-icon">solid<wbr>Logic<wbr>Singleton</a>
					</li>
					<li class=" tsd-kind-variable tsd-parent-kind-module">
						<a href="logic.html#store" class="tsd-kind-icon">store</a>
					</li>
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-variable"><span class="tsd-kind-icon">Variable</span></li>
				<li class="tsd-kind-function"><span class="tsd-kind-icon">Function</span></li>
				<li class="tsd-kind-function tsd-has-type-parameter"><span class="tsd-kind-icon">Function with type parameter</span></li>
				<li class="tsd-kind-type-alias"><span class="tsd-kind-icon">Type alias</span></li>
				<li class="tsd-kind-type-alias tsd-has-type-parameter"><span class="tsd-kind-icon">Type alias with type parameter</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-interface"><span class="tsd-kind-icon">Interface</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-class"><span class="tsd-kind-icon">Class</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="../assets/js/main.js"></script>
</body>
</html>