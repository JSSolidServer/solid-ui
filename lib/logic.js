"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.store = exports.solidLogicSingleton = exports.profile = exports.kb = exports.chat = exports.authn = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var debug = _interopRequireWildcard(require("./debug"));

var _authSession = _interopRequireDefault(require("./authn/authSession"));

var _solidLogic = require("solid-logic");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

// This module of solid-ui has a main quadstore for the app to use
//
function thisFetch(_x, _x2) {
  return _thisFetch.apply(this, arguments);
}

function _thisFetch() {
  _thisFetch = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(url, requestInit) {
    var omitCreds;
    return _regenerator["default"].wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            omitCreds = requestInit && requestInit.credentials && requestInit.credentials === 'omit';

            if (!(_authSession["default"].info.webId && !omitCreds)) {
              _context.next = 5;
              break;
            }

            return _context.abrupt("return", _authSession["default"].fetch(url, requestInit));

          case 5:
            return _context.abrupt("return", window.fetch(url, requestInit));

          case 6:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  }));
  return _thisFetch.apply(this, arguments);
}

var solidLogicSingleton = new _solidLogic.SolidLogic({
  fetch: thisFetch
}, _authSession["default"]); // Make this directly accessible as it is what you need most of the time

exports.solidLogicSingleton = solidLogicSingleton;
var store = solidLogicSingleton.store;
exports.store = store;
var kb = store; // Very commonly used synonym of store - Knowledge Base

exports.kb = kb;
var authn = solidLogicSingleton.authn;
exports.authn = authn;
var chat = solidLogicSingleton.chat; // export const language = solidLogicSingleton.language // Does not work

exports.chat = chat;
var profile = solidLogicSingleton.profile;
exports.profile = profile;
debug.log('Unique quadstore initialized.'); // ends
//# sourceMappingURL=logic.js.map