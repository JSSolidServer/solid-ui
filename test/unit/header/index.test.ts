import { silenceDebugMessages } from '../../helpers/setup'
import {
  initHeader,
  createLoginSignUpButtons,
  createUserMenuButton,
  createUserMenuLink,
  createUserMenuItem,
  rebuildHeader,
  getProfileImg,
  createBanner,
  createUserMenu,
  createHelpMenu
} from '../../../src/header'
import { NamedNode } from 'rdflib'
// @ts-ignore
import widgets from '../../../src/widgets/index'
import { solidLogicSingleton } from '../../../src/logic'

const store = solidLogicSingleton.store

silenceDebugMessages()

describe('header', () => {
  const options = { helpMenuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
  it('is exposed on public API', async () => {
    const menu = [{ label: 'Testing', onclick: () => 'https://reflectechblog.wordpress.com/' }]
    expect(await initHeader(store, menu, options)).toMatchSnapshot()
  })
})
describe('createLoginSignUpButtons', () => {
  it('creates div', () => {
    expect(createLoginSignUpButtons()).toMatchSnapshot()
  })
  it('assigns class header-banner__login', () => {
    const signUpDiv = createLoginSignUpButtons()
    expect(signUpDiv.className).toContain('header-banner__login')
  })
})
describe('createUserMenuButton', () => {
  it('creates a button', () => {
    const label = 'testing button'
    const onclick = () => { window.alert('test') }
    expect(createUserMenuButton(label, onclick)).toMatchSnapshot()
  })
  it('checks to see if the class is header-user-menu__button', () => {
    const label = 'testing button'
    const onclick = () => { window.alert('test') }
    const button = createUserMenuButton(label, onclick)
    expect(button.className).toContain('header-user-menu__button')
  })
})

describe('createUserMenuLink', () => {
  it('creates a link', () => {
    const label = 'testing link'
    const url = 'http://www.test.com'
    expect(createUserMenuLink(label, url)).toMatchSnapshot()
  })
  it('assigns link class header-user-menu__link', () => {
    const label = 'testing link'
    const url = 'http://www.test.com'
    const menuLink = createUserMenuLink(label, url)
    expect(menuLink.className).toContain('header-user-menu__link')
  })
})
describe('createUserMenuItem', () => {
  it('creates a link', () => {
    const ulElement = document.createElement('ul')
    expect(createUserMenuItem(ulElement)).toMatchSnapshot()
  })
  it('assigns list item class header-user-menu__list-item', () => {
    const ulElement = document.createElement('ul')
    const listItem = createUserMenuItem(ulElement)
    expect(listItem.className).toContain('header-user-menu__list-item')
  })
})
describe('createUserMenu', () => {
  it('creates a menu....', () => {
    const pod = new NamedNode('https://sharonstrats.inrupt.net/profile/card#me')
    const menuList = [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }]
    expect(createUserMenu(store, pod, menuList)).toMatchSnapshot()
  })
  it('assigns list item class header-banner__user-menu and header-user-menu', async () => {
    const pod = new NamedNode('https://sharonstrats.inrupt.net/profile/card#me')
    const menuList = [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }]
    const userMenu = await createUserMenu(store, pod, menuList)
    expect(userMenu.className).toContain('header-banner__user-menu')
  })
})

describe('createHelpMenu', () => {
  it('creates a menu....', () => {
    expect(createHelpMenu()).toMatchSnapshot()
  })
  it('assigns list item class header-banner__user-menu and header-user-menu', () => {
    const options = { helpMenuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
    const helpMenu = createHelpMenu(options, options.helpMenuList)
    expect(helpMenu.className).toContain('header-banner__user-menu')
  })
})
describe('rebuildHeader', () => {
  it('creates a link', () => {
    const header = document.createElement('nav')
    const pod = new NamedNode('https://test.com')
    const menuList = [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }]
    const options = { helpMenuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
    expect(rebuildHeader(header, store, pod, menuList, options)).toMatchSnapshot()
  })
})
describe('createBanner', () => {
  it('creates a link', async () => {
    const pod = new NamedNode('https://test.com')
    const user = new NamedNode('https://sharonstrats.inrupt.net/profile/card#me')
    const menuList = [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }]
    const options = { helpMenuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
    expect(await createBanner(store, pod, user, menuList, options)).toMatchSnapshot()
  })
  it('assigns list item class header-banner', async () => {
    const pod = new NamedNode('https://test.com')
    const user = new NamedNode('https://sharonstrats.inrupt.net/profile/card#me')
    const menuList = [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }]
    const options = { helpMenuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
    const banner = await createBanner(store, pod, user, menuList, options)
    expect(banner.className).toContain('header-banner')
  })
  it('check customized logo...', async () => {
    const pod = new NamedNode('https://test.com')
    const user = new NamedNode('https://sharonstrats.inrupt.net/profile/card#me')
    const menuList = [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }]
    const options = { logo: 'https://solidproject.org/assets/img/solid-emblem.svg', helpMenuList: [{ label: 'Testing', url: 'https://reflectechblog.wordpress.com/' }] }
    expect(await createBanner(store, pod, user, menuList, options)).toMatchSnapshot()
    // check something else re logo expect(banner.className).toContain('header-banner')
  })
})

// getProfileImg, need to mock widgets.findImage(user)
describe('getProfileImg', () => {
  it.skip('returns an empty image', () => {
    // eslint-disable-next-line no-undef
    // eslint-disable-next-line no-undef
    const findImageSpy = jest.spyOn(widgets, 'findImage').mockImplementation(() => { return 'test' })
    const user = new NamedNode('http://test.com')
    expect(getProfileImg(store, user)).toMatchSnapshot()
    expect(findImageSpy).toBeCalled()
  })
})
