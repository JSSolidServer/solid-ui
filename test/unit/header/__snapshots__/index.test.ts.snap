// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBanner check customized logo... 1`] = `
<div
  class="header-banner-0-1-84"
>
  <a
    class="header-banner__link-0-1-75"
    href="https://test.com"
  >
    <img
      class="header-banner__icon-0-1-76"
      src="https://solidproject.org/assets/img/solid-emblem.svg"
    />
  </a>
  <div
    class="header-banner__right-menu-0-1-85"
  >
    <div
      class="header-banner__user-menu-0-1-82 header-user-menu-0-1-83"
    >
      <button
        class="header-user-menu__trigger-0-1-81"
        type="button"
      >
        

        <svg
          fill="none"
          height="26"
          viewBox="0 0 26 26"
          width="26"
          xmlns="http://www.w3.org/2000/svg"
        >
          
    
          <path
            clip-rule="evenodd"
            d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z"
            fill="#D8D8D8"
            fill-rule="evenodd"
            stroke="#8B8B8B"
          />
          
    
          <mask
            height="26"
            id="mask0"
            mask-type="alpha"
            maskUnits="userSpaceOnUse"
            width="26"
            x="0"
            y="0"
          >
            
        
            <path
              clip-rule="evenodd"
              d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z"
              fill="white"
              fill-rule="evenodd"
              stroke="white"
            />
            
    
          </mask>
          
    
          <g
            mask="url(#mask0)"
          >
            
        
            <path
              clip-rule="evenodd"
              d="M17.0468 10.4586C17.0468 14.4979 15.4281 16.9214 12.9999 16.9214C10.5718 16.9214 8.95298 14.4979 8.95298 10.4586C8.95298 6.41931 12.9999 6.41931 12.9999 6.41931C12.9999 6.41931 17.0468 6.41931 17.0468 10.4586ZM4.09668 23.3842C6.52483 17.7293 12.9999 17.7293 12.9999 17.7293C12.9999 17.7293 19.475 17.7293 21.9031 23.3842C21.9031 23.3842 17.8481 25 12.9999 25C8.15169 25 4.09668 23.3842 4.09668 23.3842Z"
              fill="#8B8B8B"
              fill-rule="evenodd"
            />
            
    
          </g>
          

        </svg>
      </button>
      <nav
        aria-hidden="true"
        class="header-user-menu__navigation-menu-0-1-80"
      >
        <ul
          class="header-user-menu__list-0-1-77"
        >
          <li
            class="header-user-menu__list-item-0-1-79"
          >
            <a
              class="header-user-menu__link-0-1-78"
              href="https://reflectechblog.wordpress.com/"
            />
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="header-banner__user-menu-0-1-90 header-user-menu-0-1-91"
    >
      <nav
        aria-hidden="true"
        class="header-user-menu__navigation-menu-0-1-89"
      >
        <ul
          class="header-user-menu__list-0-1-86"
        >
          <li
            class="header-user-menu__list-item-0-1-88"
          >
            <a
              class="header-user-menu__link-0-1-87"
              href="https://reflectechblog.wordpress.com/"
            />
          </li>
        </ul>
      </nav>
      <button
        class="header-user-menu__trigger-0-1-92"
        type="button"
      >
        <img
          class="header-banner__help-icon-0-1-93"
          src="https://solidos.github.io/solid-ui/src/icons/https://solidos.github.io/solid-ui/src/icons/noun_help.svg"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`createBanner creates a link 1`] = `
<div
  class="header-banner-0-1-46"
>
  <a
    class="header-banner__link-0-1-37"
    href="https://test.com"
  >
    <img
      class="header-banner__icon-0-1-38"
      src="https://solidproject.org/assets/img/solid-emblem.svg"
    />
  </a>
  <div
    class="header-banner__right-menu-0-1-47"
  >
    <div
      class="header-banner__user-menu-0-1-44 header-user-menu-0-1-45"
    >
      <button
        class="header-user-menu__trigger-0-1-43"
        type="button"
      >
        

        <svg
          fill="none"
          height="26"
          viewBox="0 0 26 26"
          width="26"
          xmlns="http://www.w3.org/2000/svg"
        >
          
    
          <path
            clip-rule="evenodd"
            d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z"
            fill="#D8D8D8"
            fill-rule="evenodd"
            stroke="#8B8B8B"
          />
          
    
          <mask
            height="26"
            id="mask0"
            mask-type="alpha"
            maskUnits="userSpaceOnUse"
            width="26"
            x="0"
            y="0"
          >
            
        
            <path
              clip-rule="evenodd"
              d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z"
              fill="white"
              fill-rule="evenodd"
              stroke="white"
            />
            
    
          </mask>
          
    
          <g
            mask="url(#mask0)"
          >
            
        
            <path
              clip-rule="evenodd"
              d="M17.0468 10.4586C17.0468 14.4979 15.4281 16.9214 12.9999 16.9214C10.5718 16.9214 8.95298 14.4979 8.95298 10.4586C8.95298 6.41931 12.9999 6.41931 12.9999 6.41931C12.9999 6.41931 17.0468 6.41931 17.0468 10.4586ZM4.09668 23.3842C6.52483 17.7293 12.9999 17.7293 12.9999 17.7293C12.9999 17.7293 19.475 17.7293 21.9031 23.3842C21.9031 23.3842 17.8481 25 12.9999 25C8.15169 25 4.09668 23.3842 4.09668 23.3842Z"
              fill="#8B8B8B"
              fill-rule="evenodd"
            />
            
    
          </g>
          

        </svg>
      </button>
      <nav
        aria-hidden="true"
        class="header-user-menu__navigation-menu-0-1-42"
      >
        <ul
          class="header-user-menu__list-0-1-39"
        >
          <li
            class="header-user-menu__list-item-0-1-41"
          >
            <a
              class="header-user-menu__link-0-1-40"
              href="https://reflectechblog.wordpress.com/"
            />
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="header-banner__user-menu-0-1-52 header-user-menu-0-1-53"
    >
      <nav
        aria-hidden="true"
        class="header-user-menu__navigation-menu-0-1-51"
      >
        <ul
          class="header-user-menu__list-0-1-48"
        >
          <li
            class="header-user-menu__list-item-0-1-50"
          >
            <a
              class="header-user-menu__link-0-1-49"
              href="https://reflectechblog.wordpress.com/"
            />
          </li>
        </ul>
      </nav>
      <button
        class="header-user-menu__trigger-0-1-54"
        type="button"
      >
        <img
          class="header-banner__help-icon-0-1-55"
          src="https://solidos.github.io/solid-ui/src/icons/https://solidos.github.io/solid-ui/src/icons/noun_help.svg"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`createHelpMenu creates a menu.... 1`] = `
<div
  class="header-banner__user-menu-0-1-25 header-user-menu-0-1-26"
>
  <nav
    aria-hidden="true"
    class="header-user-menu__navigation-menu-0-1-24"
  >
    <ul
      class="header-user-menu__list-0-1-23"
    />
  </nav>
  <button
    class="header-user-menu__trigger-0-1-27"
    type="button"
  >
    <img
      class="header-banner__help-icon-0-1-28"
      src="https://solidos.github.io/solid-ui/src/icons/https://solidos.github.io/solid-ui/src/icons/noun_help.svg"
    />
  </button>
</div>
`;

exports[`createLoginSignUpButtons creates div 1`] = `
<div
  class="header-banner__login-0-1-1"
>
  <div>
    <div
      class="SolidSignInOrSignUpBox"
      style="display: flex;"
    >
      <input
        style="padding: 1em; border-radius:0.5em; font-size: 100%;background-color: #eef;"
        type="button"
        value="Log in"
      />
      <input
        style="padding: 1em; border-radius:0.5em; font-size: 100%;background-color: #efe;"
        type="button"
        value="Sign Up for Solid"
      />
    </div>
  </div>
</div>
`;

exports[`createUserMenu creates a menu.... 1`] = `
<div
  class="header-banner__user-menu-0-1-14 header-user-menu-0-1-15"
>
  <button
    class="header-user-menu__trigger-0-1-13"
    type="button"
  >
    

    <svg
      fill="none"
      height="26"
      viewBox="0 0 26 26"
      width="26"
      xmlns="http://www.w3.org/2000/svg"
    >
      
    
      <path
        clip-rule="evenodd"
        d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z"
        fill="#D8D8D8"
        fill-rule="evenodd"
        stroke="#8B8B8B"
      />
      
    
      <mask
        height="26"
        id="mask0"
        mask-type="alpha"
        maskUnits="userSpaceOnUse"
        width="26"
        x="0"
        y="0"
      >
        
        
        <path
          clip-rule="evenodd"
          d="M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z"
          fill="white"
          fill-rule="evenodd"
          stroke="white"
        />
        
    
      </mask>
      
    
      <g
        mask="url(#mask0)"
      >
        
        
        <path
          clip-rule="evenodd"
          d="M17.0468 10.4586C17.0468 14.4979 15.4281 16.9214 12.9999 16.9214C10.5718 16.9214 8.95298 14.4979 8.95298 10.4586C8.95298 6.41931 12.9999 6.41931 12.9999 6.41931C12.9999 6.41931 17.0468 6.41931 17.0468 10.4586ZM4.09668 23.3842C6.52483 17.7293 12.9999 17.7293 12.9999 17.7293C12.9999 17.7293 19.475 17.7293 21.9031 23.3842C21.9031 23.3842 17.8481 25 12.9999 25C8.15169 25 4.09668 23.3842 4.09668 23.3842Z"
          fill="#8B8B8B"
          fill-rule="evenodd"
        />
        
    
      </g>
      

    </svg>
  </button>
  <nav
    aria-hidden="true"
    class="header-user-menu__navigation-menu-0-1-12"
  >
    <ul
      class="header-user-menu__list-0-1-9"
    >
      <li
        class="header-user-menu__list-item-0-1-11"
      >
        <a
          class="header-user-menu__link-0-1-10"
          href="https://reflectechblog.wordpress.com/"
        />
      </li>
    </ul>
  </nav>
</div>
`;

exports[`createUserMenuButton creates a button 1`] = `
<button
  class="header-user-menu__button-0-1-3"
/>
`;

exports[`createUserMenuItem creates a link 1`] = `
<li
  class="header-user-menu__list-item-0-1-7"
>
  <ul />
</li>
`;

exports[`createUserMenuLink creates a link 1`] = `
<a
  class="header-user-menu__link-0-1-5"
  href="http://www.test.com"
/>
`;

exports[`getProfileImg returns an empty image 1`] = `
"
<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"26\\" height=\\"26\\" viewBox=\\"0 0 26 26\\" fill=\\"none\\">
    <path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z\\" fill=\\"#D8D8D8\\" stroke=\\"#8B8B8B\\"/>
    <mask id=\\"mask0\\" mask-type=\\"alpha\\" maskUnits=\\"userSpaceOnUse\\" x=\\"0\\" y=\\"0\\" width=\\"26\\" height=\\"26\\">
        <path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M13 25C19.6274 25 25 19.6274 25 13C25 6.37258 19.6274 1 13 1C6.37258 1 1 6.37258 1 13C1 19.6274 6.37258 25 13 25Z\\" fill=\\"white\\" stroke=\\"white\\"/>
    </mask>
    <g mask=\\"url(#mask0)\\">
        <path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M17.0468 10.4586C17.0468 14.4979 15.4281 16.9214 12.9999 16.9214C10.5718 16.9214 8.95298 14.4979 8.95298 10.4586C8.95298 6.41931 12.9999 6.41931 12.9999 6.41931C12.9999 6.41931 17.0468 6.41931 17.0468 10.4586ZM4.09668 23.3842C6.52483 17.7293 12.9999 17.7293 12.9999 17.7293C12.9999 17.7293 19.475 17.7293 21.9031 23.3842C21.9031 23.3842 17.8481 25 12.9999 25C8.15169 25 4.09668 23.3842 4.09668 23.3842Z\\" fill=\\"#8B8B8B\\"/>
    </g>
</svg>"
`;

exports[`header is exposed on public API 1`] = `undefined`;

exports[`rebuildHeader creates a link 1`] = `[Function]`;
